server/src/services/duplicate.service.ts

```diff
@@ -15,8 +15,20 @@ import { usePagination } from 'src/utils/pagination';
 export class DuplicateService extends BaseService {
   async getDuplicates(auth: AuthDto): Promise<DuplicateResponseDto[]> {
     const res = await this.assetRepository.getDuplicates({ userIds: [auth.user.id] });
-
-    return mapDuplicateResponse(res.map((a) => mapAsset(a, { auth, withStack: true })));
+    const uniqueAssetIds: string[] = [];
+    const duplicates = mapDuplicateResponse(res.map((a) => mapAsset(a, { auth, withStack: true }))).filter(
+      (duplicate) => {
+        if (duplicate.assets.length === 1) {
+          uniqueAssetIds.push(duplicate.assets[0].id);
+          return false;
+        }
+        return true;
+      },
+    );
+    if (uniqueAssetIds.length > 0) {
+      void this.assetRepository.updateAll(uniqueAssetIds, { duplicateId: null });
```

Comment:
Why are you not awaiting this?

Reply:
My reasoning is that the update operation is non-critical to the response, so it doesn’t need to hold up the function's main purpose, which is to return valid duplicates.

Reply:
Not awaiting the promise will make catching a potential error impossible though. So, if the update should fail for whatever reason, the process is f'ed

Reply:
I see. 
Would a try catch block be the correct way of handling this?

Reply:
No. You need to await it. It's not like this is gonna take long either way though, so whatever